### DISCUSSION TITLE
Is it possible to download my entire Watch Later playlist?

### DISCUSSION THREAD

**Discussion by Proaxel** [Tyrrrz/YoutubeDownloader#207] [Q&A]
My watch later playlist is huge, I would like to download it so I can watch them all on flights and other times where I don't have a good internet connection. It looks like watch later is always private with no way to change it and thus the downloader app can't access it. Is there a way to make it public or if not is there some sort of workaround?

**Comment by Tyrrrz** [comment]
You can probably try to temporarily copy videos from watch later playlist to another publicly available playlist:
https://webapps.stackexchange.com/questions/39686/how-to-copy-youtube-playlist-from-one-account-to-another

I don't know if the answer in that thread is still relevant, it was given 7 years ago.

**Reply by Proaxel** [reply]
Yeah, the comments on the top say it doesn't work anymore, the latest comment in December of 2020 saying it doesn't work. 
![image](https://user-images.githubusercontent.com/9545347/117555202-f4a02280-b011-11eb-9f01-6352116d534e.png)


I guess the only way is to add them one by one to another playlist and then give the downloader the URL of that?

**Reply by Tyrrrz** [reply]
If you can somehow just get a list of all URLs that works too, you can copy paste multiple URLs at once in YTD. One thing you can do is open the playlist, scroll to the very end until all videos load and then use dev console to get all video IDs.

```javascript
console.log([...document.getElementsByTagName("a")].filter(e => e.id === "video-title").map(e => e.href).join("\n"))
```

![image](https://user-images.githubusercontent.com/1935960/117576858-41304000-b09c-11eb-8ccb-a7f20ca27d8e.png)

**Reply by Tyrrrz** [reply]
Maybe you will need to trim off the `&list` part of the URL so that YoutubeDownloader won't try to parse it as playlist though. You can just search and replace `&list` with something else.

**Reply by Tyrrrz** [reply]
To clarify, to open your watch later playlist and use the snippet above, go to https://www.youtube.com/playlist?list=WL

**Reply by Proaxel** [reply]
(Excuse the late reply, I'm very bad at responding to stuff)

I copied and pasted the list into Notepad++ and used Regex to delete everything after the &list, then I copied and pasted that into the app. It took a while for the app to load the video list, but it all worked okay for the most part. Thanks!
